### PR TITLE
#1659 Autoload next video - replace content provider with WebResource…

### LINKED ISSUE #1659: Autoload next video
*@kelson42 commented on Dec 17, 2019, 3:09 PM UTC:*

I thought this would be a nice feature and by testing the ZIM about Mali I discovered that it was somehow working halfway. At the end of the video it seems that new one is played... but the HTML page does not change... it is buggy!

*This issue was moved by [kelson42](https://github.com/kelson42) from [openzim/youtube#60](https://github.com/openzim/youtube/issues/60).*

### PR DESCRIPTION
**Pull request by macgills**
… input streams

Fixes #1659 

We supply Zim resources via input streams to the WebView, removing the need for the content provider.
This is not to be considered complete until we are able to read the length of articles from kiwixlib as longs

### CODE REVIEW

**Comment by kelson42** [comment]
@macgills nice ...

File: core/src/main/java/org/kiwix/kiwixmobile/core/JNIInitialiser.kt [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+      icuDir.absolutePath
+    } catch (e: Exception) {
+      Log.w(TAG_KIWIX, "Error copying icu data file", e)
+      // TODO: Consider surfacing to user
+      null
+    }
+  }
+
+  fun init() {
+    //  empty method so class is not reported as unused
```
Comment (abdulwd) [inline core/src/main/java/org/kiwix/kiwixmobile/core/JNIInitialiser.kt:59]: Why can't we move the `init` block to this function?

Comment (macgills) [reply core/src/main/java/org/kiwix/kiwixmobile/core/JNIInitialiser.kt:59]: The init block was more expressive and I didn't have to store context/JniKiwix in variables, might be better to just suppress the unused warning and delete the init fun

Comment (abdulwd) [reply core/src/main/java/org/kiwix/kiwixmobile/core/JNIInitialiser.kt:59]: Suppressing the warning and adding a doc string explaining why this class is being highlighted as unused will be great.

Comment (macgills) [reply core/src/main/java/org/kiwix/kiwixmobile/core/JNIInitialiser.kt]: done

**Comment by kelson42** [comment]
New kiwix-lib is now available https://bintray.com/kiwix/kiwix/kiwixlib/9.1.1

**Pull request merged**